src/main.rs

```diff
@@ -264,14 +266,14 @@ fn get_dest_from_to(to: &str) -> Result<Box<dyn Write>> {
     if to == "-" {
```

Comment:
We could have `dest = if to == "-" {` ... here like you did for backend above.

I sometimes find the statement based style easier to read than the expression based one but happy to conform with the latter if that's preferred.

Reply:
So did I, but you get used to it. It's worth I think, because sometimes, it reveals some other refactor that is even more idiomatic.

At this point I believe that for almost every algorithm, there is only one idiomatic representation in Rust. And every non-idiomatic expression has something that can be refactored until you reach the idiomatic one :D It's probably not true, but it's nice to believe that you've reached the best code possible for what you are trying to do.